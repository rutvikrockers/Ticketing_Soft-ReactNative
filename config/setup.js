// create our store for testing redux components
import configureMockStore from 'redux-mock-store';
const thunk = require('redux-thunk');

// Setup adapter to work with enzyme 3.2.0
const Enzyme = require('enzyme');
const Adapter = require('enzyme-adapter-react-16');

export const mockStore = configureMockStore([thunk.default]);

Enzyme.configure({ adapter: new Adapter() });