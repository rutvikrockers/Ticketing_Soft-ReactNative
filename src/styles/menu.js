import colors from './colors';

const menu = {
  menu: {
    padding: 10,
    paddingTop: 60,
    alignItems: 'stretch'
  },
  menuItem: {
    padding: 20,
    paddingBottom: 0,
    fontSize: 18
  },
  menuHeader: {
    padding: 20,
    marginTop: 20,
    fontSize: 18,
    fontStyle: 'italic'
  }
}

export default menu;
