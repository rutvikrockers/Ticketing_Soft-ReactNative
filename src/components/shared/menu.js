import React, { Component } from 'react';
import {
  AppRegistry,
  StyleSheet,
  Text,
  FlatList,
  TouchableOpacity,
  View
} from 'react-native';
import {Link, Redirect} from 'react-router-native';
import {connect} from 'react-redux';
import {destroySession} from '../../actions';
import {playVideo, isVideoPlaying} from '../../actions/video';
import Button from './button';

import styles from '../../styles/main.js';

class Menu extends Component {
  constructor(props) {
    super(props);
  }
  logout() {
    this.props.dispatch(destroySession()); 
  }
  componentWillMount(){
    this.props.dispatch(isVideoPlaying(false));
    this.props.dispatch(playVideo(0)) 
  }
  render() {
    if(!this.props.loggedIn){
      return (<Redirect to="/" />)
    }
    return (
      
      <View style={styles.root}>
        <View style={styles.header}>
          <Link to="/"><Text>Back</Text></Link>
          <Text style={styles.headerText} >Menu</Text>
        </View>
        <View style={styles.menu}>
            
          <Text style={styles.menuHeader}>{this.props.user.email}</Text>
          <Link to="/profile"><Text style={styles.menuItem}>Profile</Text></Link>
         
          <Text
            style={styles.menuItem}
            onPress={() => {this.logout()}}
          >Logout</Text>
        </View>
      </View>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    user: state.user,
    loggedIn: state.user.loggedIn
  }
}

export default connect(mapStateToProps)(Menu);
