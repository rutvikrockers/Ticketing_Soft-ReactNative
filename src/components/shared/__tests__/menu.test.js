import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { mockStore } from '../../../../config/setup';
import toJson from 'enzyme-to-json';
import { default as Menu } from '../menu';

const render = (props) => shallow(
    <Provider store={mockStore({})}>
        <Menu {...props} />
    </Provider>
);

describe('Menu', () => {
    let component;
    beforeEach(() => {
        component = render({});
    });

    it('matches its snapshot', () => {
        expect(toJson(component, { mode: 'deep' })).toMatchSnapshot();
    });

    it('renders', () => {
        expect(component.find(Menu).length).toBe(1);
    });
});