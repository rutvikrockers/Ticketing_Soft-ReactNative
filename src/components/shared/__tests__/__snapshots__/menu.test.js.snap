// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu matches its snapshot 1`] = `<Connect(Menu) />`;
